### DISCUSSION TITLE
How to redraw after toggle group_empty_dir option?

### DISCUSSION THREAD

**Discussion by weilbith** [nvim-neo-tree/neo-tree.nvim#1249] [Q&A] [answered]
Hey 👋🏾 

I like to use the option to `group_empty_dir`. But sometimes I need to access a grouped directory like to add a new folder there or similar. So I added this custom command in my configuration:

```lua
require('neo-tree').setup({
  -- ...
  commands = {
    toggle_group_empty_dir = function(state)
      state.group_empty_dir = not state.group_empty_dir
    end
  }
})
```

This kinda works. In terms that this alternates the rendering/behavior for folders I have not opened yet. But for already grouped empty directories, it does not ungroup them. I tried to add a `require('neo-tree.ui.renderer').redraw(state)`, but that doesn't help.
How can I make to update it? Properly?

---

**Visualisation**

Actual filesystem:
```
- one/
   - two/
      - file.ext
- three/
  - four/
    - five/
      - file.ext
  - file.ext
```

So my initial view, having nothing unfold yet and having empty dirs grouped, it looks like that:

```
- one/two/
- three/
```

If I would open folder `three` still with grouping, it looks like that:

```
- one/two/
- three/
  - four/five/
  - file.ext
```

If I reset now to the initial state and before opening folder `three` call my toggle function, nothing changes first. So the `one/two` is still grouped. But if I now open folder `three`, its children are ungrouped. So I just see:

```
- one/two/
- three/
  - four/
  - file.ext
```

**Comment by cseickel** [comment, marked as answer]
What you need there is a `refresh` instead of a `redraw`. `refresh` actually recreates the nodes, but `redraw` just updates the display of those nodes.

```
require("neo-tree.sources.manager").refresh(state.name)
```

**Reply by weilbith** [reply]
Amazing! Thank you very much 🤗
